src/plugins/event-plugins/ResourcePlugin.ts

```diff
@@ -15,90 +15,72 @@ import {
+        this.eventCount = 0;
+        this.resourceObserver = new PerformanceObserver(
+            this.performanceEntryHandler
+        );
     }
 
     enable(): void {
         if (this.enabled) {
             return;
         }
         this.enabled = true;
-        window.addEventListener(LOAD, this.resourceEventListener);
+        this.resourceObserver.observe({
+            type: RESOURCE,
+            buffered: true
+        });
     }
 
     disable(): void {
         if (!this.enabled) {
             return;
         }
         this.enabled = false;
-        if (this.resourceEventListener) {
-            window.removeEventListener(LOAD, this.resourceEventListener);
-        }
+        this.resourceObserver.disconnect();
     }
 
-    resourceEventListener = (event: Event): void => {
+    performanceEntryHandler = (list: PerformanceObserverEntryList): void => {
+        this.recordPerformanceEntries(list.getEntries());
+    };
+
+    recordPerformanceEntries = (list: PerformanceEntryList) => {
         const recordAll: PerformanceEntry[] = [];
         const sample: PerformanceEntry[] = [];
-        let eventCount = 0;
-
-        const resourceObserver = new PerformanceObserver((list) => {
-            list.getEntries()
-                .filter((e) => e.entryType === RESOURCE)
-                .filter((e) => !this.config.ignore(e))
-                .forEach((event) => {
-                    // Out of n resource events, x events are recorded using Observer API
-                    const type: ResourceType = getResourceFileType(event.name);
-                    if (this.config.recordAllTypes.includes(type)) {
-                        recordAll.push(event);
-                    } else if (this.config.sampleTypes.includes(type)) {
-                        sample.push(event);
-                    }
-                });
-        });
-        resourceObserver.observe({
-            entryTypes: [RESOURCE]
-        });
 
-        // Remaining (n-x) resource events are recorded using getEntriesByType API.
-        // Note: IE11 browser does not support Performance Observer API. Handle the failure gracefully
-        const events = performance.getEntriesByType(RESOURCE);
-        if (events !== undefined && events.length > 0) {
-            events
-                .filter((e) => !this.config.ignore(e))
-                .forEach((event) => {
-                    const type: ResourceType = getResourceFileType(event.name);
-                    if (this.config.recordAllTypes.includes(type)) {
-                        recordAll.push(event);
-                    } else if (this.config.sampleTypes.includes(type)) {
-                        sample.push(event);
-                    }
-                });
-        }
+        list.filter((e) => e.entryType === RESOURCE)
+            .filter((e) => !this.config.ignore(e))
+            .forEach((event) => {
+                const type: ResourceType = getResourceFileType(event.name);
```

Comment:
I have found that sometimes event.name does not include the correct file type extension. Could we add logic to use `initatorType` as a backup method to verify this information?

Reply:
https://github.com/aws-observability/aws-rum-web/pull/448/files#diff-7f9eb043b568ee17c6e1abb8ce54754a6f6a1571604e94509f756b5a751a6e1cR62-R63

Reply:
I agree using `initiatorType` sounds like a good idea to get more accurate resource types. That is, instead of using only the file extension to identify the resource type,`getResourceFileType` would also use `initiatorType`.

I believe this is a new feature though. In #448, could you also use `getResourceFileType`? Then we can make a separate PR to improve `getResourceFileType` accuracy by considering `initatorType`.